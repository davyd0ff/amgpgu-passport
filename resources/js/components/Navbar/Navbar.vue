<template>
  <header>
    <div class="navbar-fixed">
      <nav class="navbar teal">
        <div class="nav-wrapper">
          <a class="white-text" href="#">{{ title | localize }}</a>
          <a class="sidenav-trigger" href="#" data-target="sidenav-left">
            <i class="material-icons">menu</i>
          </a>
          <ul id="nav-mobile" class="right">
            <notifications-drop-down />
          </ul>
        </div>
      </nav>
    </div>
  </header>
</template>

<script>
import NotificationsDropDown from "@/components/NotificationsDropDown";

export default {
  name: "Navbar",
  components: { NotificationsDropDown },
  computed: {
    title: function () {
      if (typeof this.$route.meta.title === "function") {
        return this.$route.meta.title(this.$route);
      }
      return this.$route.meta?.title || "";
    },
  },
};
</script>

<style scoped></style>
