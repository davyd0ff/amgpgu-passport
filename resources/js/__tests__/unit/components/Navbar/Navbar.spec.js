import { shallowMount, createLocalVue } from '@vue/test-utils';
import Navbar from '@/components/Navbar/Navbar.vue';

const localize = (str) => str;
const localVue = createLocalVue();
localVue.filter('localize', localize);

describe('Navbar.vue', () => {
  it('render', () => {
    const $route = {
      meta: {
        title: 'TEST TITLE',
      },
    };

    const wrapper = shallowMount(Navbar, {
      localVue,
      mocks: { $route },
    });

    expect(wrapper.element).toMatchSnapshot();
    // expect(wrapper.find('.brand-logo').innerText).toBe('TEST TITLE');
  });
});
