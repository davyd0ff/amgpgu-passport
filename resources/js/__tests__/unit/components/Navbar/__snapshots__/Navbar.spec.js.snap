// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue render 1`] = `
<header>
  <div
    class="navbar-fixed"
  >
    <nav
      class="navbar teal"
    >
      <div
        class="nav-wrapper"
      >
        <a
          class="white-text"
          href="#"
        >
          TEST TITLE
        </a>
         
        <a
          class="sidenav-trigger"
          data-target="sidenav-left"
          href="#"
        >
          <i
            class="material-icons"
          >
            menu
          </i>
        </a>
         
        <ul
          class="right"
          id="nav-mobile"
        >
          <notifications-drop-down-stub />
        </ul>
      </div>
    </nav>
  </div>
</header>
`;
